boot/zephyr/Kconfig

```diff
@@ -364,6 +370,50 @@ config BOOT_WAIT_FOR_USB_DFU
 	  for USB DFU to be invoked. Please note DFU always updates the
 	  slot1 image.
 
+config BOOT_USB_DFU_WAIT_DELAY_MS
+	int
+	depends on BOOT_USB_DFU_WAIT
+	default 12000
+	help
+	  The duration to wait for USB DFU to be invoked.
+
+config BOOT_USB_DFU_GPIO
+	bool "Use GPIO to detect whether to trigger DFU mode"
+	select USB
+	select USB_DFU_CLASS
+	select IMG_MANAGER
+	help
+	  If y, MCUboot uses GPIO to detect whether to invoke USB DFU.
+	  Please note DFU always updates the slot1 image.
+
+endchoice
+
+config BOOT_USB_DFU_DETECT_PORT
+	string "GPIO device to trigger USB DFU mode"
+	depends on BOOT_USB_DFU_GPIO
```

Comment:
You may enclose all of the configs that depend on `BOOT_USB_DFU_GPIO` in if/endif block (https://docs.zephyrproject.org/latest/guides/kconfig/tips.html?highlight=kconfig#factoring-out-common-dependencies)